sources/src/PeerPlugin.cpp

```diff
@@ -241,6 +241,20 @@ namespace extension {
             return true;
         }
 
+        // Only communicate with peers with same protocol major version number
+        if(_protocolVersionOfPeer.major() != protocol_statemachine::CBStateMachine::protocolVersion.major()) {
+          // Mark peer as not supporting this extension
+          _peerPaymentBEPSupportStatus = BEPSupportStatus::not_supported;
+
+          // Remove peer
+          std::clog << "Dropping Peer: (incompatible protocol vesrion)" << std::endl;
+          libtorrent::error_code ec;
+          drop(ec);
```

Comment:
I think this is a bit strict, its not clear that we cannot hold a connection open for normal bittorrent/DHT traffic. It could also be useful for the extension user to understand how often and what peers are using incompatible protocol version. To this end, one could introduce a new enum value in `BEPSupportStatus`  which reflects an incompatible version explicitly. For now though, this is good enough I think, so at best making an issue out of this could be interesting, and perhaps marking it with "good first issue" or help wanted.